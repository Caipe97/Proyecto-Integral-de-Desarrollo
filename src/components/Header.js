import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import logo from '../images/logo_small.png';
import { Link } from "react-router-dom";

class Header extends Component {
  render() {
    console.log(this.props)
    return (
      <div className="header" style={{ background: '#0E4749', paddingTop: '10px', paddingBottom: '10px', top: 0 }}>
        <nav className="navbar sticky-top navbar-expand-lg navbar-dark p-3">
          <div className="container">
            <img src={logo} alt="imagen tracking" style={{ width: 200 }} />
            {this.props.userId > 0
              ? <div className="collapse navbar-collapse" id="navbarNav" style={{height:'23px'}}>
                <ul className="navbar-nav ms-auto">
                  <li>
                    <Link name='link' style={{color: 'white', textDecoration: 'none'}} onClick={() => {
                      this.props.onLogout();
                      this.props.history.push("/");
                    }
                    }>Cerrar Sesi√≥n</Link>
                  </li>
                </ul>
              </div>
              : null}
          </div>
        </nav>
      </div>
    )
  }
}

export default Header;