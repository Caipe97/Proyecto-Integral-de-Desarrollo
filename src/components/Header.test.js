import React from 'react';
import { shallow } from 'enzyme'; 
import Header from './Header';

describe('Header tests', () => {
    const historyMock = { push: jest.fn() };
    const mockPropsSignIn = {userId: 1, onLogout: jest.fn()};
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Header {...mockPropsSignIn} history={historyMock}/>);
    })

    it('expect to render Header component to render with signing in', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('expect to logout and go to login when pressing the button', () => {
        wrapper.find('[type="button"]').at(0).simulate('click');
        expect(wrapper.instance().props.onLogout).toHaveBeenCalledTimes(1);
        expect(wrapper.instance().props.history.push).toHaveBeenCalledTimes(1);
        expect(historyMock.push.mock.calls[0]).toEqual(['/']);
    })
})
