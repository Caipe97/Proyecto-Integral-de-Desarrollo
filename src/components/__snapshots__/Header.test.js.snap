// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests expect to render Header component to render with signing in 1`] = `
<div
  className="header"
  style={
    Object {
      "background": "#0E4749",
      "paddingBottom": "10px",
      "paddingTop": "10px",
      "top": 0,
    }
  }
>
  <nav
    className="navbar sticky-top navbar-expand-lg navbar-dark p-3"
  >
    <div
      className="container"
    >
      <img
        alt="imagen tracking"
        src="logo_small.png"
        style={
          Object {
            "width": 200,
          }
        }
      />
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
        style={
          Object {
            "height": "23px",
          }
        }
      >
        <ul
          className="navbar-nav ms-auto"
        >
          <li>
            <Link
              name="link"
              onClick={[Function]}
              style={
                Object {
                  "color": "white",
                  "textDecoration": "none",
                }
              }
            >
              Cerrar Sesi√≥n
            </Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
